feat(flutter): improve app start docs (#11891)
getsentry/sentry-docs: docs/platforms/flutter/integrations/app-start-instrumentation.mdx

Sentry's app start instrumentation provides insight into how long your application takes to launch.

<Note>

App start instrumentation is available on **iOS** and **Android**.

</Note>

## Instrumentation Behaviour

Before diving into the configuration, it's important to understand how app start instrumentation behaves:

> App start instrumentation tracks the duration between the earliest native process initialization and the first frame rendered (as reported by [addTimingsCallback](https://api.flutter.dev/flutter/scheduler/SchedulerBinding/addTimingsCallback.html)). Once the app start is processed, we remove the callback to avoid additional overhead.

Reviewer: ```suggestion
App start instrumentation tracks the duration between the earliest native process initialization and the first frame rendered (as reported by [addTimingsCallback](https://api.flutter.dev/flutter/scheduler/SchedulerBinding/addTimingsCallback.html)). Once the app start is processed, the callback is removed to avoid additional overhead.
```

Reviewer: I think we should avoid the "we" here because it implies that the user may need to do something (contrary to how we recommend using it [here](https://develop.sentry.dev/engineering-practices/documentation/#the-reader--the-author) for steps the user actually needs to perform).

@coolguyzone I tried to find guidance for this in our [contribution docs](https://docs.sentry.io/contributing/approach/#writing-for-sentry), but didn't find anything - do you know if we have general recommendations? is it worth writing it down somewhere?

Reviewer 2: Sorry I missed this comment, @kahest, generally the docs are written in the second person (e.g. 'you can install the Sentry SDK by . . .') but I do think it is a good idea to be explicit here and either say 'the callback is removed' or 'Sentry removes the callback' to avoid any confusion. I'll see if there's a way to incorporate this into the style guide.

> When the SDK receives the start and end times of the app launch, we:

Reviewer: similar to above, avoid "we" here unless the user should do something - please check the rest of the PR for similar occurrences

## Configure

This instrumentation is automatically enabled. There is no need for further configuration.

## Verify

### 1. Launch Your App:

Launch your Sentry configured app.

### 2. Locate Your Transaction:

> Open the [sentry.io](https://sentry.io/) performance dashboard, find, and select the 'root /' transaction and navigate to the trace view of a sampled event.

Reviewer: ```suggestion
Open the [sentry.io performance page](https://sentry.io/performance), find, and select the 'root /' transaction and navigate to the trace view of a sampled event.
```

### 3. View App Start Metrics:

> Select the event within your transaction. Sentry.io displays the app start metrics on the right side of the screen in the **Mobile Vitals** section.

Reviewer: ```suggestion
Select the event within your transaction. Sentry displays the app start metrics on the right side of the screen in the **Mobile Vitals** section.
```
